services/cd-service/pkg/service/git.go

```diff
@@ -201,12 +201,21 @@ func (s *GitServer) GetCommitInfo(ctx context.Context, in *api.GetCommitInfoRequ
 	}
 	sort.Strings(touchedApps)
 	var events []*api.Event
+	loadMore := false
 	if s.OverviewService.Repository.State().DBHandler.ShouldUseOtherTables() {
 		events, err = db.WithTransactionMultipleEntriesT(s.OverviewService.Repository.State().DBHandler, ctx, true, func(ctx context.Context, transaction *sql.Tx) ([]*api.Event, error) {
-			return s.GetEvents(ctx, transaction, fs, commitPath)
+			return s.GetEvents(ctx, transaction, fs, commitPath, pageNumber, pageSize)
 		})
+		if len(events) > int(pageSize) {
+			loadMore = true
+			events = events[:len(events)-1]
+		}
 	} else {
-		events, err = s.GetEvents(ctx, nil, fs, commitPath)
+		events, err = s.GetEvents(ctx, nil, fs, commitPath, pageNumber, pageSize)
+		if len(events) > int(pageSize) {
+			loadMore = true
+		}
+		events = events[pageNumber:min(len(events), int(pageNumber+pageSize))]
```

Comment:
this line should be part of `GetEvents`.
Why are we starting at `pageNumber`? It would make more sense to me if it was `pageNumber * pageSize`.
Something like this:
```suggestion
		events = events[pageNumber*pageSize:min(len(events), int((pageNumber+1)*pageSize))]
```